src/Strategy/Python/Poetry/Common.hs

```diff
@@ -0,0 +1,184 @@
+      PyProjectPoetryGitDependencySpec _ -> GitType
+      PyProjectPoetryUrlDependencySpec _ -> URLType
+      _ -> PipType
+
+    depName = case deps of
+      PoetryTextVersion _ -> name
+      PyProjectPoetryDetailedVersionDependencySpec _ -> name
+      PyProjectPoetryGitDependencySpec ds -> gitUrl ds
+      PyProjectPoetryUrlDependencySpec ds -> sourceUrl ds
+      PyProjectPoetryPathDependencySpec ds -> sourcePath ds
+
+    depVersion = case deps of
+      PoetryTextVersion ds -> toDependencyVersion ds
+      PyProjectPoetryDetailedVersionDependencySpec ds -> toDependencyVersion (poetryDependencyVersion ds)
+      PyProjectPoetryGitDependencySpec ds -> case asum [gitTag ds, gitRev ds, gitBranch ds] of
+        Nothing -> Nothing
+        Just version -> Just $ CEq version
+      _ -> Nothing
+
+    depEnvironment = depEnvs
+    depLocations = []
+    depTags = Map.empty
+
+-- | Maps poetry lock package to map of package name and associated dependency.
+toMap :: [PoetryLockPackage] -> Map.Map PackageName Dependency
+toMap pkgs = Map.fromList $ (\x -> (lowerCasePkgName x, toDependency x)) <$> (filter supportedPoetryLockDep pkgs)
+  where
+    lowerCasePkgName :: PoetryLockPackage -> PackageName
+    lowerCasePkgName pkg = PackageName (toLower $ unPackageName $ poetryLockPackageName pkg)
+
+    toDependency :: PoetryLockPackage -> Dependency
+    toDependency pkg =
+      Dependency
+        { dependencyType = toDepType (poetryLockPackageSource pkg)
+        , dependencyName = toDepName pkg
+        , dependencyVersion = toDepVersion pkg
+        , dependencyLocations = toDepLocs pkg
+        , dependencyEnvironments = toDepEnvironment pkg
+        , dependencyTags = Map.empty
+        }
+
+    toDepName :: PoetryLockPackage -> Text
+    toDepName lockPkg = case (poetryLockPackageSource lockPkg) of
+      Nothing -> unPackageName $ poetryLockPackageName lockPkg
+      Just lockPkgSrc -> case poetryLockPackageSourceType lockPkgSrc of
+        "legacy" -> unPackageName $ poetryLockPackageName lockPkg
+        _ -> poetryLockPackageSourceUrl lockPkgSrc
+
+    toDepType :: Maybe PoetryLockPackageSource -> DepType
+    toDepType Nothing = PipType
+    toDepType (Just lockPkgSrc) = case poetryLockPackageSourceType lockPkgSrc of
+      "git" -> GitType
+      "url" -> URLType
+      _ -> PipType
+
+    toDepLocs :: PoetryLockPackage -> [Text]
+    toDepLocs pkg = case poetryLockPackageSource pkg of
+      Nothing -> []
+      Just lockPkgSrc -> case poetryLockPackageSourceType lockPkgSrc of
+        "legacy" -> [poetryLockPackageSourceUrl lockPkgSrc]
+        _ -> []
+
+    toDepVersion :: PoetryLockPackage -> Maybe VerConstraint
+    toDepVersion pkg = Just $
+      CEq $
+        fromMaybe (poetryLockPackageVersion pkg) $ do
+          lockPkgSrc <- poetryLockPackageSource pkg
+          ref <- poetryLockPackageSourceReference lockPkgSrc
+          if poetryLockPackageSourceType lockPkgSrc /= "legacy" then Just ref else Nothing
```

Comment:
[optional nit]: Since you unconditionally wrap in a `Just` here, the return type can probably be `a` instead of `Maybe a`.  You could move the `Just` constructor to the caller, and then this function will do less (often good for code organization)